pennylane/_grad.py

```diff
@@ -21,55 +21,101 @@
 from autograd.numpy.numpy_boxes import ArrayBox
 from autograd.extend import vspace
 from autograd.wrap_util import unary_to_nary
+from pennylane.compiler import compiler
 
 make_vjp = unary_to_nary(_make_vjp)
 
 
 class grad:
-    """Returns the gradient as a callable function of (functions of) QNodes.
+    """A :func:`~.qjit` compatible gradient transformation that returns the gradient
+    as a callable function of (functions of) QNodes.
 
-    By default, gradients are computed for arguments which contain the property
-    ``requires_grad=True``. Alternatively, the ``argnum`` keyword argument can
-    be specified to compute gradients for function arguments without this property,
+
+    By default, in interpreted mode, gradients are computed for arguments which contain
+    the property ``requires_grad=True``. Alternatively, the ``argnum`` keyword argument
+    can be specified to compute gradients for function arguments without this property,
     such as scalars, lists, tuples, dicts, or vanilla NumPy arrays. Setting
     ``argnum`` to the index of an argument with ``requires_grad=False`` will raise
     a ``NonDifferentiableError``.
 
     When the output gradient function is executed, both the forward pass
-    *and* the backward pass will be performed in order to
-    compute the gradient. The value of the forward pass is available via the
-    :attr:`~.forward` property.
+    *and* the backward pass will be performed in order to compute the gradient.
+    The value of the forward pass is available via the :attr:`~.forward` property.
 
     .. warning::
         ``grad`` is intended to be used with the Autograd interface only.
 
+    .. note::
+
+        When used with :func:`~.qjit`, this function only supports the Catalyst compiler.
+        Please see :func:`catalyst.grad` for more details.
```

Comment:
That's a good point 👍

Reply:
This was marked as resolved @maliasadi - what was the resolution?

> When used with [qjit()](https://xanaduai-pennylane--4692.com.readthedocs.build/en/4692/code/api/pennylane.qjit.html), this function only supports the Catalyst compiler. Please see [catalyst.grad()](https://docs.pennylane.ai/projects/catalyst/en/stable/code/api/catalyst.grad.html#catalyst.grad) for more details.

This part means little to me. What other compiler could I choose?

> Please see the Catalyst [quickstart guide](https://docs.pennylane.ai/projects/catalyst/en/stable/dev/quick_start.html), as well as the [sharp bits and debugging tips](https://docs.pennylane.ai/projects/catalyst/en/stable/dev/sharp_bits.html) page for an overview of the differences between Catalyst and PennyLane.

This part seems fine, perhaps it could be part of a single simple note box that informs the user about `qjit`.

Reply:
> what was the resolution?

it's fixed in PR #4812.

> This part means little to me. What other compiler could I choose?

currently, there's only Catalyst. But we're planning to explore the addition of more compilers in near future. I don't think it's harmful to add this sentence until we extend the support. This also make users curious to check on the qjit and compiler docs for future updates. I'm happy to change it in anyways to better convey the message. cc: @josh146

Reply:
Yep, it is deliberately vague in the sense that we don't have additional compilers rn, but we are planning to add some.

Reply:
Chatted with @trbromley offline. There's no concern with this one anymore.